File: src/main/java/com/ormComp/model/Building.java
package com.ormComp.model;

import jakarta.persistence.*;
import lombok.Data;
import lombok.Getter;

import java.util.List;
import java.util.Objects;

@Entity
@Getter
@Table(name = "building", schema = "rental")
public class Building {
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Id
    private long id;
    @Column(name = "description")
    private String description;
    @ManyToOne
    @JoinColumn(name = "owner_id")
    private User owner;
    @ManyToOne
    @JoinColumn(name = "address_id")
    private Address address;
    @OneToMany(mappedBy = "building")
    private List<Flat> flat;
}

File: DotNet/Models/Building.cs
using System;
using System.Collections.Generic;

namespace Endpoints.Models;

public partial class Building
{
    public long Id { get; set; }

    public long AddressId { get; set; }

    public string? Description { get; set; }

    public long OwnerId { get; set; }

    public virtual Address Address { get; set; } = null!;

    public virtual ICollection<Flat> Flats { get; set; } = new List<Flat>();

    public virtual User Owner { get; set; } = null!;
}
